### PR TITLE
feat: implement normalization cache for non-persisted operations

### PR DESCRIPTION
**Pull request by jensneuse**

### CODE REVIEW

File: router/core/operation_processor.go [hunk 1]
```diff
@@ -227,6 +256,8 @@ func (o *OperationKit) Parse(ctx context.Context, clientInfo *ClientInfo) error
 		if err != nil {
 			return err
 		}
+		// it's important to make a copy of the persisted operation data, because it's used in the cache
+		// we might modify it later, so we don't want to modify the cached data
 		o.parsedOperation.Request.Query = string(persistedOperationData)
```
Comment (StarpTech) [inline router/core/operation_processor.go:261]: `unsafebytes.BytesToString(key)`

File: router/pkg/config/config.schema.json [hunk 2]
```diff
@@ -1271,7 +1275,7 @@
         },
         "execution_plan_cache_size": {
           "type": "integer",
-          "default": 10000,
+          "default": 1024,
```
Comment (StarpTech) [inline router/pkg/config/config.schema.json:1278]: Changing default can be unpredictable for customers. How likely is it that this can have an impact?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/core/operation_processor.go (hunk 1 above)
Comment (jensneuse) [reply router/core/operation_processor.go:261]: Did you read the comment? above :D

File: router/pkg/config/config.schema.json (hunk 2 above)
Comment (jensneuse) [reply router/pkg/config/config.schema.json:1278]: 1024 plans cached should be more than enough for most setups. I don't expect any negative impacts.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**